Some parse_op_by_op_results.py QOL updates (git info, totals, conditonal formatting)

kmabeeTT opened the pull request:
### Ticket
None

### What's changed
 - Move summary worksheet creation to function since it's growing, and create few helper functions
 - Add more totals for Compiled-to-TTNN and Executed-On-Device, and various conditional color formatting
 - See comment in PR for example of what it looks like before/after

### Checklist
- [ ] New/Existing tests provide coverage for changes

kmabeeTT commented:
![Screenshot 2025-03-20 at 3 33 12 PM](https://github.com/user-attachments/assets/cc06eb18-c77a-47a5-ad0d-5aaa3ce46b40)

Here is comparison before/after with things as they stand right now (local subset of test reports downloaded from a recently nightly)

jameszianxuTT commented on results/parse_op_by_op_results.py:
```diff
@@ -175,6 +198,201 @@ def create_test_dirs():
+    info_str = (
+        f"Generated on {date_str} at {time_str} from Branch: {branch} Commit: {commit}"
+    )
+    worksheet.write(0, 0, info_str)
```
nit: Putting this in 0,0 and then offsetting all the data in the actual spreadsheet might not be the best way to embed such metadata. This might have been better to embed in a separate worksheet, or even in the [document properties](https://xlsxwriter.readthedocs.io/example_doc_properties.html)

Dealing with this offset and the hardcoding of row/col indices will mean that future bulk adjustments of this data within this worksheet summary page are going to be pretty painful (eg. if you did want to reverse this and not embed the info_str and move all the actual data to 0,0, you'd have to change pretty much every index reference in this function). But adding enums for row/col data or computing row/col indices dynamically might be overengineering this; 

I guess it just depends on your vision for how much you'd want to modify this worksheet in the future

kmabeeTT replied:
Thanks a lot - yeah, it's a valid concern, I feel the same way. Did want this particular line up front so it's easy to share a screenshot and compare screenshots side by side, but yeah, the column/row maintenance is not ideal.  Going to keep it for now, and maybe in future revisit with a cleanup task (feel free if you have a good idea there too).

kmabeeTT commented:
Fixed a small bug in the script (Total Ops per Compile Depth included the first row still, made it use a variable to prevent accidental including of wrong range, rebased/squashed/force-pushed).
